src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -238,11 +255,15 @@ public override byte[] Sign(byte[] input)
             {
                 throw new ObjectDisposedException(GetType().ToString());
             }
-
+#if DNXCORE50
+            if (rsaCng != null)
+                return rsaCng.SignData(input, hash, RSASignaturePadding.Pkcs1);
```

Comment:
SignData is available on the RSA base class; you really should avoid casting to the implementation type unless you're doing native interop.